core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -5095,7 +5095,7 @@ public boolean isSubjectToMandatoryReadPermissionCheck(String restOfPath) {
     }
 
     private boolean isAgentJnlpPath(String restOfPath, String prefix) {
-        return restOfPath.matches("/computer/[^/]+/" + prefix + "-agent[.]jnlp");
+        return restOfPath.matches("(?:/manage)?/computer/[^/]+/" + prefix + "-agent[.]jnlp");
```

Comment:
```suggestion
        return restOfPath.matches("(/manage)?/computer/[^/]+/" + prefix + "-agent[.]jnlp");
```

Non-capturing group seems irrelevant?